safe_transaction_service/history/serializers.py

```diff
@@ -569,10 +569,26 @@ def validate(self, attrs):
                 SafeSignatureType.ETH_SIGN,
             ):
                 raise ValidationError("Only EOA and ETH_SIGN signatures are supported")
+
+            # The transaction can be deleted by the proposer or by the delegate user who proposed it.
             if safe_signature.owner == proposer:
                 return attrs
+            if (
+                multisig_tx.proposed_by_delegate
+                and safe_signature.owner == multisig_tx.proposed_by_delegate
```

Comment:
IIRC `safe_signature.owner` involves computation. If you are using in multiple places please assign it to a variable first

Reply:
Makes sense. Fixed [here](https://github.com/safe-global/safe-transaction-service/pull/2294/commits/3b80d157ad862d16ccae2182bbdacae4b0f3cec1)